kubectl-fdb/cmd/cordon.go

```diff
@@ -141,42 +140,24 @@ func cordonNode(cmd *cobra.Command, kubeClient client.Client, inputClusterName s
 			errors = append(errors, internalErr)
 			continue
 		}
+		var podNames []string
+		for _, pod := range pods.Items {
+			podNames = append(podNames, pod.Name)
+		}
 
-		clusterNames := getClusterNames(cmd, inputClusterName, pods, clusterLabel)
-		for clusterName := range clusterNames {
-			cmd.Printf("Starting operation on %s, node: %s\n", clusterName, node)
-			cluster, err := loadCluster(kubeClient, namespace, clusterName)
-			if err != nil {
-				internalErr := fmt.Sprintf("unable to load cluster: %s, skipping\n", clusterName)
-				errors = append(errors, internalErr)
-				cmd.PrintErr(internalErr)
-				continue
-			}
-
-			var processGroups []string
-			for _, pod := range pods.Items {
-				// With the field selector above this shouldn't be required, but it's good to
-				// have a second check.
-				if pod.Spec.NodeName != node {
-					cmd.Printf("Pod: %s is not running on node %s will be ignored\n", pod.Name, node)
-					continue
-				}
-
-				if internal.ContainsPod(cluster, pod) {
-					processGroup, ok := pod.Labels[cluster.GetProcessGroupIDLabel()]
-					if !ok {
-						cmd.Printf("could not fetch process group ID from Pod: %s\n", pod.Name)
-						continue
-					}
-					processGroups = append(processGroups, processGroup)
-				}
-			}
-			err = replaceProcessGroups(kubeClient, cluster.Name, processGroups, namespace, "", "", withExclusion, wait, false, true)
-			if err != nil {
-				internalErr := fmt.Sprintf("unable to cordon all Pods for cluster %s\n", cluster.Name)
-				errors = append(errors, internalErr)
-				cmd.PrintErr(internalErr)
-			}
+		cmd.Printf("Cordoning node: %s\n", node)
+		err = replaceProcessGroups(cmd, kubeClient, inputClusterName, podNames, namespace, replaceProcessGroupsOptions{
```

Comment:
note: this swap would result in a behavioral difference: a failure to patch any of the clusters (or if the user says no) would have it error, rather than the current behavior which is to collate all the errors and have the user retry.   I think we should pick one behavior and have the node logic follow it (as-is, the node logic collates removal errors of non-collated cluster removal errors)

Reply:
That change makes sense and simplifies the code 👍 Exiting early is also fine as the user can rerun the command again with the same result, e.g. the process group IDs will be added to the removal list.